pandas/core/window/rolling.py

```diff
@@ -619,6 +621,8 @@ def _apply(
         )
 
         result.index = result_index
+        if not self._as_index:
+            result = result.reset_index(level=list(range(len(groupby_keys))))
```

Comment:
What happens here when the groupby is on an explicit list, e.g. in your test use `groupby(["A", "A", "B", "B"])` instead. What is groupby_keys in this case?

Reply:
Here's the result

```
In [4]: df.groupby(["A", "A", "B", "B"], as_index=False).rolling(window=2, min_periods=1).mean()
> /Users/matthewroeschke/pandas-mroeschke/pandas/core/window/rolling.py(580)_apply()
-> result_index_names = groupby_keys + grouped_index_name
(Pdb) groupby_keys
[None]
(Pdb) c
Out[4]:
                num
  date
A 2018-01-01  100.0
  2018-01-02  150.0
B 2018-01-01  150.0
  2018-01-02  200.0

In [5]: df.groupby(["A", "A", "B", "B"], as_index=False).mean()
Out[5]:
     num
0  150.0
1  200.0
```

it appears in either case the grouping list isn't mapped as a column on the dataframe